### PR TITLE
Lower kt destructuring declarations with var RHS

### PR DESCRIPTION
**Pull request by ursachec**

### CODE REVIEW

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala [hunk 1]
```diff
@@ -1326,6 +1329,184 @@ class AstCreator(fileWithMeta: KtFileWithMeta, xTypeInfoProvider: TypeInfoProvid
...
+  def astsForDestructuringDeclarationWithVarRHS(
+    expr: KtDestructuringDeclaration,
+    scopeContext: ScopeContext,
+    order: Int
+  )(implicit fileInfo: FileInfo, typeInfoProvider: TypeInfoProvider): Seq[AstWithCtx] = {
+    val typedInit = {
+      expr.getInitializer match {
+        case typed: KtNameReferenceExpression => Some(typed)
+        case _                                => None
+      }
```
Comment (mpollmeier) [inline joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala:1359]:
```suggestion
      Option(expr.getInitializer).filter(_.isInstanceOf[KtNameReferenceExpression])
```
matter of taste...

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala [hunk 2]
```diff
@@ -1326,6 +1329,184 @@ class AstCreator(fileWithMeta: KtFileWithMeta, xTypeInfoProvider: TypeInfoProvid
...
+          }.headOption
+
+        val matchingRefOption =
+          if (matchingLocal.isDefined) {
+            Some(matchingLocal.get)
+          } else if (matchingMethodParam.isDefined) {
+            Some(matchingMethodParam.get)
+          } else {
+            None
+          }
```
Comment (mpollmeier) [inline joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala:1454]:
```suggestion
        val matchingRefOption = matchingLocal.orElse(matchingMethodParam)
```

**Review by mpollmeier — APPROVED** [review]

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala (hunk 2 above)
Comment (ursachec) [reply joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala:1454]: ah, nice, thx!

File: joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala (hunk 1 above)
Comment (ursachec) [reply joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/passes/AstCreator.scala]: nicer, danke schoen!

**Review by max-leuthaeuser — APPROVED** [review]

**Pull request merged**